File: backend/microservice/src/player/api/internal/admin/CreateShopItemInternalApi.scala
package microservice.player.api.internal.admin

import cats.effect.IO
import java.sql.Connection
import microservice.infrastructure.api.{APIMessage, PlanSteps}
import microservice.infrastructure.http.HttpError
import microservice.player.objects.shop.ShopItem
import microservice.player.support.shop.ShopCatalogSupport

/** 模块间 API：创建商店商品；由 admin HTTP API 调用，不挂路由。 */
final case class CreateShopItemInternalAPIMessage(
  name: String,
  description: String,
  price: Int,
  currency: String,
  catalogIndex: Int,
  active: Boolean,
  sortOrder: Int
) extends APIMessage[ShopItem] {
  override def plan(connection: Connection): IO[Either[HttpError, ShopItem]] =
    PlanSteps.finish {
      PlanSteps.read(
        ShopCatalogSupport.createItem(
          connection,
          name,
          description,
          price,
          currency,
          catalogIndex,
          active,
          sortOrder
        )
      )
    }
}

File: frontend/src/api/player/internal/admin/CreateShopItemInternalApi.ts
import { z } from "zod";
import { APIMessage } from "../../../../system/api/APIMessage.js";

/** Mirrors backend internal API player/api/internal/admin/CreateShopItemInternalApi.scala; not registered as a public frontend route. */
export class CreateShopItemInternalAPI extends APIMessage<unknown> {
  readonly name: unknown;
  readonly description: unknown;
  readonly price: unknown;
  readonly currency: unknown;
  readonly catalogIndex: unknown;
  readonly active: unknown;
  readonly sortOrder: unknown;
  constructor(name: unknown, description: unknown, price: unknown, currency: unknown, catalogIndex: unknown, active: unknown, sortOrder: unknown) {
    super();
    this.name = name;
    this.description = description;
    this.price = price;
    this.currency = currency;
    this.catalogIndex = catalogIndex;
    this.active = active;
    this.sortOrder = sortOrder;
  }

  override get responseSchema() {
    return z.unknown();
  }
}
